src/views/catalog/customize/components/CustomizeSource/CustomizeSource.tsx

```diff
@@ -0,0 +1,176 @@
+  }, []);
+
+  React.useEffect(() => {
+    switch (selectedSourceType) {
+      case DEFAULT_SOURCE:
+        return onChange(undefined);
+      case PVC_SOURCE_NAME:
+        return onChange(getPVCSource(pvcNameSelected, pvcNamespaceSelected));
+      case HTTP_SOURCE_NAME:
+        return onChange(getGenericSourceCustomization(selectedSourceType, httpURL, volumeQuantity));
+      case REGISTRY_SOURCE_NAME:
+        return onChange(
+          getGenericSourceCustomization(selectedSourceType, containerImage, volumeQuantity),
+        );
+    }
+  }, [
+    onChange,
+    pvcNameSelected,
+    pvcNamespaceSelected,
+    httpURL,
+    containerImage,
+    volumeQuantity,
+    selectedSourceType,
+  ]);
+
+  return (
+    <>
+      <FormGroup
+        label={<SelectDiskSourceLabel />}
+        fieldId={`disk-source-required-disk`}
+        isRequired
+        className="disk-source-form-group"
+      >
+        <Select
+          isOpen={isOpen}
+          onToggle={setIsOpen}
+          onSelect={onSelectDiskSource}
+          variant={SelectVariant.single}
+          selections={selectedSourceType}
+        >
+          <SelectOption
+            value={DEFAULT_SOURCE}
+            description={t('Use the default template disk source')}
+          >
+            <span data-test-id={DEFAULT_SOURCE}>{t('Default')}</span>
+          </SelectOption>
+          <SelectOption
+            value={PVC_SOURCE_NAME}
+            description={t(
+              'Select an existing persistent volume claim already available on the cluster and clone it.',
+            )}
+          >
+            <span data-test-id={PVC_SOURCE_NAME}>{t('PVC (creates PVC)')}</span>
+          </SelectOption>
+
+          <SelectOption
+            value={HTTP_SOURCE_NAME}
+            description={t('Import content via URL (HTTP or S3 endpoint).')}
+          >
+            <span data-test-id={HTTP_SOURCE_NAME}>{t('URL (creates PVC)')}</span>
+          </SelectOption>
+
+          <SelectOption
+            value={REGISTRY_SOURCE_NAME}
+            description={t('Import content via container registry.')}
+          >
+            <span data-test-id={REGISTRY_SOURCE_NAME}>{t('Registry (creates PVC)')}</span>
+          </SelectOption>
+        </Select>
+      </FormGroup>
+
+      {selectedSourceType === PVC_SOURCE_NAME && (
+        <PersistentVolumeClaimSelect
+          pvcNameSelected={pvcNameSelected}
+          projectSelected={pvcNamespaceSelected}
+          selectNamespace={selectPVCNamespace}
+          selectPVCName={selectPVCName}
+        />
+      )}
+
+      {selectedSourceType === HTTP_SOURCE_NAME && (
+        <FormGroup
+          label={t('Image URL')}
+          fieldId={`disk-source-required-${selectedSourceType}`}
+          isRequired
+          className="disk-source-form-group"
+        >
+          <TextInput
+            value={httpURL}
+            type="text"
+            onChange={setHTTPURL}
+            aria-label={t('Image URL')}
```

Comment:
we need to have better validation here. at least `startsWith('https://...)` cc @yaacov

Reply:
Okay for me. Is there also for the container URL some simple validation that we can use?

Reply:
the validation is "nice to have", the requirement is to have a text input.

lets start with something working, and then after we have it working add the validation, in the next step